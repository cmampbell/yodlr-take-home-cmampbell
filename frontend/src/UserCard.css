.card {
    height: 100px;
    width: 400px;
    margin-left: 20px;
    margin-bottom: 15px;
}

.card-body {
    padding:0%;
}

.card-header {
    width: 400px;
    padding: 0%;

    background-color: #fcfcfc;
    text-align: left;
    text-indent: 5%;
}

.active {
    border-color: green;
}

.pending {
    border-color: #8F8547;
}

.card-text {
    text-align: left;
    text-indent: 5%;
}