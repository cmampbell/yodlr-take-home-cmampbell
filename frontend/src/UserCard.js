import React from "react";
import { Card, CardBody, CardText, CardHeader } from 'reactstrap'
import './UserCard.css'

const UserCard = ({ user }) => {
    const { firstName, lastName, email, state } = user

    return (
        <Card color="light" outline >
            <CardBody>
                <CardHeader className={state}>
                    {firstName} {lastName}
                </CardHeader>
                <CardText>
                    Email: {email}
                </CardText>
                <CardText>
                    State: {state.charAt(0).toUpperCase() + state.slice(1)}
                </CardText>
            </CardBody>
        </Card>
    )
}

export default UserCard;