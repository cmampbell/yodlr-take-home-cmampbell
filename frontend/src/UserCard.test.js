import React from "react";
import renderWithRouter from "./renderWithRouter";
import UserCard from "./UserCard";

const testUser = {firstName: 'bilbo', lastName: 'baggins', email: 'thehobbit@middleearth.com', state: 'active'}
describe('NavBar tests', () => {

    it('should render without crashing', () => {
        // define a full route as the first argument
        renderWithRouter(<UserCard user={testUser}/>)
    })

    it('should match the snapshot', ()=> {
        const { asFragment } = renderWithRouter(<UserCard user={testUser}/>)

        expect(asFragment()).toMatchSnapshot();
    })
})