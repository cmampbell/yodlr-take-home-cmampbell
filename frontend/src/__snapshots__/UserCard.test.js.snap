// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar tests should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="card border-light"
  >
    <div
      class="card-body"
    >
      <div
        class="active card-header"
      >
        bilbo baggins
      </div>
      <p
        class="card-text"
      >
        Email: thehobbit@middleearth.com
      </p>
      <p
        class="card-text"
      >
        State: Active
      </p>
    </div>
  </div>
</DocumentFragment>
`;
